pyvista/core/filters/data_set.py

```diff
@@ -6853,8 +6853,8 @@ def shrink(self, shrink_factor=1.0, progress_bar=False):
         >>> shrunk.plot(show_edges=True, line_width=5)
 
         """
-        if not (0.0 <= shrink_factor <= 1.0):
-            raise ValueError('`shrink_factor` should be between 0.0 and 1.0')
+        _validation.check_type(shrink_factor, float)
+        _validation.check_range(shrink_factor, rng=[0.0, 1.0])
```

Comment:
`check_type` will probably fail for numpy float scalars
```suggestion
        _validation.check_dtype(shrink_factor, float)
        _validation.check_range(shrink_factor, rng=[0.0, 1.0])
```

alternatively: 

```suggestion
        shrink_factor = _validation.validate_number(
            shrink_factor,
            must_have_dtype=float, 
            must_be_in_range=[0.0, 1.0],
        )
```

Reply:
Thanks!